### ISSUE TITLE
Is it intentional that a Pod with readinessProbe that's continuously erroring will still have Ready = True?

### ISSUE THREAD

**Issue by davidxia** [kubernetes/kubernetes#74732]
**What happened**:

tl;dr: Only probes that fail will actually cause the pod's ready state to be changed. Is it intentional that a Pod with readinessProbe that's continuously erroring will still have `Ready = True`?

A probe by the kubelet can end in one of three states:

- successful
- failed (command returned a non-0 exit code)
- errored (command did not return before the timeout elapsed, the command does not exist inside the container, etc)

[Here is the code](https://github.com/kubernetes/kubernetes/blob/v1.10.11/pkg/kubelet/prober/prober.go#L108) where the event `probe errored` is recorded. Note that `err != nil`.

[Here is the code](https://github.com/kubernetes/kubernetes/blob/7dbe426b2426458f02339e28d1cc102efdb6eda1/pkg/kubelet/prober/worker.go#L217-L221) that calls the above function - when `err != nil` (the probe returned an error), the result is discarded. 

**What you expected to happen**:

I expect the Pod to have `Ready != True`.

**How to reproduce it (as minimally and precisely as possible)**:

`kubectl apply` the following pod

```
apiVersion: v1
kind: Pod
metadata:
  name: test-readiness-probe
  namespace: dxia
spec:
  containers:
  - name: test-container
    image: davidxia/readiness-probe-test:1 # a public image on docker hub
    command: [ "/bin/bash", "-c", "--" ]
    args: [ "while true; do sleep 30; done;" ]
    readinessProbe:
      exec:
        command:
        - bash
        - /test.sh
      failureThreshold: 3
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
```

test.sh is

```
M=$(date +%M)
if (( 0 <= 10#$M && 10#$M < 30 )); then
    cat
else
    echo SUCCESS
    exit 0
fi
```

So if you create the Pod after the 30 minute mark, it'll start and reach RUNNING. But after the 0 minute mark, the probe will start failing from timeouts. But the Pod is still `Ready=True` as shown below.

```
kubectl -n dxia describe pods test-readiness-probe

Name:         test-readiness-probe
Namespace:    dxia
Node:         gke-production-us-ce-terraform-201812-971e980a-hcr6/10.178.32.76
Start Time:   Thu, 28 Feb 2019 09:43:17 -0500
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"Pod","metadata":{"annotations":{},"name":"test-readiness-probe","namespace":"dxia"},"spec":{"containers":[{"args":["while tr...
              kubernetes.io/psp=gce.persistent-volume-binder
Status:       Running
IP:           10.179.49.92
Containers:
  test-container:
    Container ID:  docker://c746a73e1c5f570418d160a1f09fc70caabc84c60f0a490cd1994eb32157be34
    Image:         davidxia/readiness-probe-test:1
    Image ID:      docker-pullable://davidxia/readiness-probe-test@sha256:e93bf5f7e60f83e6518568bf4ecf24b6f4dd074f4583ab8ed81b757fcaf0b915
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/bash
      -c
      --
    Args:
      while true; do sleep 30; done;
    State:          Running
      Started:      Thu, 28 Feb 2019 09:43:25 -0500
    Ready:          True
    Restart Count:  0
    Readiness:      exec [bash /test.sh] delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-vlmm6 (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          True
  PodScheduled   True
Volumes:
  default-token-vlmm6:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-vlmm6
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                 Age              From                                                          Message
  ----     ------                 ----             ----                                                          -------
  Normal   Scheduled              30m              default-scheduler                                             Successfully assigned test-readiness-probe to gke-production-us-ce-terraform-201812-971e980a-hcr6
  Normal   SuccessfulMountVolume  30m              kubelet, gke-production-us-ce-terraform-201812-971e980a-hcr6  MountVolume.SetUp succeeded for volume "default-token-vlmm6"
  Normal   Pulling                30m              kubelet, gke-production-us-ce-terraform-201812-971e980a-hcr6  pulling image "davidxia/readiness-probe-test:1"
  Normal   Pulled                 30m              kubelet, gke-production-us-ce-terraform-201812-971e980a-hcr6  Successfully pulled image "davidxia/readiness-probe-test:1"
  Normal   Created                30m              kubelet, gke-production-us-ce-terraform-201812-971e980a-hcr6  Created container
  Normal   Started                30m              kubelet, gke-production-us-ce-terraform-201812-971e980a-hcr6  Started container
  Warning  Unhealthy              1m (x2 over 7m)  kubelet, gke-production-us-ce-terraform-201812-971e980a-hcr6  Readiness probe errored: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Anything else we need to know?**

Originally asked as [SO question here](https://stackoverflow.com/questions/54842616/kubernetes-pods-readinessprobe-errored-but-endpoint-not-removed-from-service).

**Environment**:
- Kubernetes version (use `kubectl version`):

Same behavior still exists on master.

```
kubectl --context gke_gke-xpn-1_us-central1_production-us-central1 version                                                                             

Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.7", GitCommit:"0c38c362511b20a098d7cd855f1314dad92c2780", GitTreeState:"clean", BuildDate:"2018-08-20T10:09:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.11-gke.1", GitCommit:"5c4fddf874319c9825581cc9ab1d0f0cf51e1dc9", GitTreeState:"clean", BuildDate:"2018-11-30T16:18:58Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"
```

- Cloud provider or hardware configuration:
GKE but doesn't really matter

**Comment by davidxia** [comment]
cc @mattnworb @tommyulfsparre

**Comment by davidxia** [comment]
not stale
